metaflow/client/core.py

```diff
@@ -877,7 +886,9 @@ def tarball(self) -> tarfile.TarFile:
         TarFile
             TarFile for everything in this code package
         """
-        return self._tar
+        if self._backend == TarPackagingBackend:
```

Comment:
safer to compare string. also avoids a TarPackagingBackend import (internal implementation detail)